src/diff_generate.c

```diff
@@ -1274,6 +1274,12 @@ int git_diff__from_iterators(
 	b_opts.flags = FLAGS_SECOND; \
 	b_opts.start = pfx; \
 	b_opts.end = pfx; \
+	if (opts && opts->range_start && *opts->range_start && \
+			(!pfx || !*pfx || strcmp(opts->range_start, pfx) > 0)) \
```

Comment:
The unconditional use of `strcmp` is incorrect. If the iterator is case-insensitive this should be `strcasecmp`. I cheated here because this makes no difference for gitstatusd.